### PR TITLE
Micro-optimization to curry

### PR DESCRIPTION
**Pull request by dalefrancis88**
adding in a propery to check if the passed in function has already been run through our curry method

### CODE REVIEW

File: src/core/curry.js [hunk 1]
```diff
@@ -5,12 +5,15 @@ const isFunction = require('./isFunction')
...
+
   return fn.length > 1 ? fn.bind(null, arg) : fn.call(null, arg)
 }
 
 // curry : ((a, b, c) -> d) -> a -> b -> c -> d
 function curry(fn) {
-  return function(...xs) {
+  if(fn.isCurried) { return fn }
+
+  const curried = function(...xs) {
```
Comment (evilsoft) [inline src/core/curry.js]: Could this become a function as opposed to assignment.

File: src/core/curry.js [hunk 2]
```diff
@@ -28,6 +31,14 @@ function curry(fn) {
 
     return val
   }
+
+  Object.defineProperty(curried, 'isCurried', {
```
Comment (evilsoft) [inline src/core/curry.js]: Lets go with `@@crocks/curried` for the signifier

File: src/core/curry.js (hunk 1 above)
Comment (dalefrancis88) [reply src/core/curry.js]: Not sure what you mean by ' function as opposed to assignment', do you mean `function curried(...xs)` ?

Comment (evilsoft) [reply src/core/curry.js]: yeppers, named function I should of said.

File: src/core/curry.js [hunk 3]
```diff
@@ -5,12 +5,15 @@ const isFunction = require('./isFunction')
...
 function applyCurry(fn, arg) {
   if(!isFunction(fn)) { return fn }
+
   return fn.length > 1 ? fn.bind(null, arg) : fn.call(null, arg)
 }
 
 // curry : ((a, b, c) -> d) -> a -> b -> c -> d
 function curry(fn) {
-  return function(...xs) {
+  if(fn['@@crocks/curried']) { return fn }
```
Comment (evilsoft) [inline src/core/curry.js]: How about we create a constant for the signal and use a multiline `if` (should probably add that to lint soon)
```js
const CURRY_SYMB =
  '@@crocks/curried'

if(fn[CURRY_SYMB]) {
  return fn
}
```

File: src/helpers/unary.js [hunk 4]
```diff
@@ -2,16 +2,15 @@
 /** @author Ian Hofmann-Hicks (evil) */
 
 const isFunction = require('../core/isFunction')
+const nAry = require('./nAry')
```
Comment (evilsoft) [inline src/helpers/unary.js]: Do not think this change belongs in this PR. Thoughts?

**Pull request merged**